homeassistant/config_entries.py

```diff
@@ -247,6 +247,14 @@ def __init__(self, version: str, domain: str, title: str, data: dict,
                               component.DOMAIN)
                 result = False
 
+            device_registry = await \
```

Comment:
This is the wrong method. Unload is not removal perse. It's just undoing the setup work.  

Removal is done here:

https://github.com/home-assistant/home-assistant/blob/fa81385b5cdbc409b683ea24406fcc22949f0f64/homeassistant/config_entries.py#L312-L330